packages/nc-gui-v2/components/smartsheet-toolbar/FieldListAutoCompleteDropdown.vue

```diff
@@ -48,14 +50,25 @@ const localValue = computed({
   },
 } */
 const options = computed<SelectProps['options']>(() =>
-  meta?.value?.columns?.map((c) => ({
-    value: c.id,
-    label: c.title,
-    icon: h(isVirtualCol(c) ? VirtualCellIcon : CellIcon, {
-      columnMeta: c,
-    }),
-    c,
-  })),
+  meta?.value?.columns
+    ?.filter(
+      (c: ColumnType) =>
+        c &&
+        (isSort
+          ? !(
+              c.uidt === UITypes.LinkToAnotherRecord &&
+              (c.colOptions as LinkToAnotherRecordType).type !== RelationTypes.BELONGS_TO
+            )
+          : !c.colOptions || !c.system),
+    )
```

Comment:
This seems very unreadable and complicated - is there a more simple or at least more readable way of writing this down? Terniary operation looks really bad here, in my opinion.

Reply:
Will replace with `if...else`